language_service/src/compilation.rs

```diff
@@ -125,13 +128,14 @@ impl Compilation {
 
     /// Regenerates the compilation with the same sources but the passed in configuration options.
     pub fn recompile(&mut self, package_type: PackageType, target_profile: TargetProfile) {
-        let sources = self.user_source_contents();
+        let sources: Vec<_> = self
+            .user_source_contents()
+            .into_iter()
+            .map(|(a, b)| (Arc::from(a), Arc::from(b)))
+            .collect();
 
         let new = match self.kind {
-            CompilationKind::OpenDocument => {
-                assert!(sources.len() == 1);
-                Self::new_open_document(sources[0].0, sources[0].1, package_type, target_profile)
-            }
+            CompilationKind::OpenDocument => Self::new(&sources, package_type, target_profile),
```

Comment:
I think you'll want to update the doc comment (and possibly name) for `OpenDocument` to reflect the fact that it applies to more than single-file projects

Reply:
Maybe just `Project`?